server/server.go

```diff
@@ -256,3 +262,182 @@ func (r *rodeServer) ListOccurrences(ctx context.Context, occurrenceRequest *pb.
+	}
+	if res.IsError() {
+		return nil, createError(log, "error searching elasticsearch for document", nil, zap.String("response", res.String()), zap.Int("status", res.StatusCode))
+	}
+
+	var searchResults esSearchResponse
+	if err := decodeResponse(res.Body, &searchResults); err != nil {
+		return nil, createError(log, "error unmarshalling elasticsearch response", err)
+	}
+
+	if searchResults.Hits.Total.Value == 0 {
+		log.Debug("document not found", zap.Any("search", search))
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("%T not found", policy))
+	}
+	protojson.Unmarshal(searchResults.Hits.Hits[0].Source, proto.MessageV2(policy))
+
+	return policy, nil
+
+}
+
+func (r *rodeServer) DeletePolicy(ctx context.Context, deletePolicyRequest *pb.DeletePolicyRequest) (*emptypb.Empty, error) {
+	log := r.logger.Named("DeletePolicy")
+
+	search := &esSearch{
+		Query: &filtering.Query{
+			Term: &filtering.Term{
+				"id.keyword": deletePolicyRequest.Id,
+			},
+		},
+	}
+
+	encodedBody, requestJSON := encodeRequest(search)
+	log.Debug("es request payload", zap.Any("payload", requestJSON))
+
+	res, err := r.esClient.DeleteByQuery(
+		[]string{rodeElasticsearchPoliciesIndex},
+		encodedBody,
+		r.esClient.DeleteByQuery.WithContext(ctx),
+	)
+	if err != nil {
+		return nil, createError(log, "error sending request to elasticsearch", err)
+	}
+	if res.IsError() {
+		return nil, createError(log, "received unexpected response from elasticsearch", nil)
+	}
+
+	var deletedResults esDeleteResponse
+	if err = decodeResponse(res.Body, &deletedResults); err != nil {
+		return nil, createError(log, "error unmarshalling elasticsearch response", err)
+	}
+
+	if deletedResults.Deleted == 0 {
+		return nil, createError(log, "elasticsearch returned zero deleted documents", nil, zap.Any("response", deletedResults))
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (r *rodeServer) ListPolicies(ctx context.Context, listPoliciesRequest *pb.ListPoliciesRequest) (*pb.ListPoliciesResponse, error) {
+	log := r.logger.Named("List Policies")
+
+	// filtering logic
+
+	// encodedBody, requestJSON := encodeRequest(search)
+	// log.Debug("es request payload", zap.Any("payload", requestJSON))
+	var policies []*pb.Policy
+	res, err := r.esClient.Search(
+		r.esClient.Search.WithContext(ctx),
+		r.esClient.Search.WithIndex(rodeElasticsearchPoliciesIndex),
+	)
+	//log = log.With(zap.String("request", requestJson))
+
+	if err != nil {
+		return nil, createError(log, "error sending request to elasticsearch", err)
+	}
+	if res.IsError() {
+		return nil, createError(log, "error searching elasticsearch for document", nil, zap.String("response", res.String()), zap.Int("status", res.StatusCode))
+	}
+
+	var searchResults esSearchResponse
+	if err := decodeResponse(res.Body, &searchResults); err != nil {
+		return nil, createError(log, "error unmarshalling elasticsearch response", err)
+	}
+
+	if searchResults.Hits.Total.Value == 0 {
```

Comment:
If we are going to follow REST standards, a call to an endpoint that is returning a collection with no entities should return an empty collection instead of `404`. This is the same "problem" that the UI is having with [occurrences and resources](https://github.com/rode/rode/issues/36) when trying to fetch the collection - there's no way to distinguish between when a collection is found but doesn't have any entities vs the endpoint is not found at all in the API.

Reply:
on it